src/source/Signaling/LwsApiCalls.c

```diff
@@ -1468,6 +1479,186 @@ STATUS connectSignalingChannelLws(PSignalingClient pSignalingClient, UINT64 time
+    UINT32 resultLen;
+
+    CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
+    CHK(pSignalingClient->channelEndpointWebrtc[0] != '\0', STATUS_INTERNAL_ERROR);
+
+    // Create the API url
+    STRCPY(url, pSignalingClient->channelEndpointWebrtc);
+    STRCAT(url, JOIN_STORAGE_SESSION_API_POSTFIX);
+
+    // Prepare the json params for the call
+    if (pSignalingClient->pChannelInfo->channelRoleType == SIGNALING_CHANNEL_ROLE_TYPE_VIEWER) {
+        SNPRINTF(paramsJson, ARRAY_SIZE(paramsJson), SIGNALING_JOIN_STORAGE_SESSION_VIEWER_PARAM_JSON_TEMPLATE,
+                 pSignalingClient->channelDescription.channelArn, pSignalingClient->clientInfo.signalingClientInfo.clientId);
+    } else {
+        SNPRINTF(paramsJson, ARRAY_SIZE(paramsJson), SIGNALING_JOIN_STORAGE_SESSION_MASTER_PARAM_JSON_TEMPLATE,
+                 pSignalingClient->channelDescription.channelArn);
+    }
+
+    // Create the request info with the body
+    CHK_STATUS(createRequestInfo(url, paramsJson, pSignalingClient->pChannelInfo->pRegion, pSignalingClient->pChannelInfo->pCertPath, NULL, NULL,
+                                 SSL_CERTIFICATE_TYPE_NOT_SPECIFIED, pSignalingClient->pChannelInfo->pUserAgent,
+                                 SIGNALING_SERVICE_API_CALL_CONNECTION_TIMEOUT, SIGNALING_SERVICE_API_CALL_COMPLETION_TIMEOUT,
+                                 DEFAULT_LOW_SPEED_LIMIT, DEFAULT_LOW_SPEED_TIME_LIMIT, pSignalingClient->pAwsCredentials, &pRequestInfo));
+
+    // createRequestInfo does not have access to the getCurrentTime callback, this hook is used for tests.
+    if (pSignalingClient->signalingClientCallbacks.getCurrentTimeFn != NULL) {
+        pRequestInfo->currentTime =
+            pSignalingClient->signalingClientCallbacks.getCurrentTimeFn(pSignalingClient->signalingClientCallbacks.customData);
+    }
+
+    checkAndCorrectForClockSkew(pSignalingClient, pRequestInfo);
+
+    CHK_STATUS(createLwsCallInfo(pSignalingClient, pRequestInfo, PROTOCOL_INDEX_HTTPS,
+                                 &pLwsCallInfo)); //!< TBD. Accroding to the design document, the prefix of url will be webrtc://
+
+    if (pSignalingClient->mediaStorageConfig.storageStatus) {
+        pSignalingClient->diagnostics.joinSessionToOfferRecvTime = GETTIME();
+    }
+    // Make a blocking call
+    CHK_STATUS(lwsCompleteSync(pLwsCallInfo));
+
+    // Set the service call result
+    ATOMIC_STORE(&pSignalingClient->result, (SIZE_T) pLwsCallInfo->callInfo.callResult);
+    pResponseStr = pLwsCallInfo->callInfo.responseData;
+    resultLen = pLwsCallInfo->callInfo.responseDataLen;
+
+    // Early return if we have a non-success result
+    CHK((SERVICE_CALL_RESULT) ATOMIC_LOAD(&pSignalingClient->result) == SERVICE_CALL_RESULT_OK, STATUS_SIGNALING_LWS_CALL_FAILED);
+
+    // Perform some validation on the channel description
+    CHK(pSignalingClient->channelDescription.channelStatus != SIGNALING_CHANNEL_STATUS_DELETING, STATUS_SIGNALING_CHANNEL_BEING_DELETED);
```

Comment:
Shouldnt we have a similar check at the start for this before we go ahead and construct the requestInfo? Why have it at the end?

Reply:
This is set after the `lwsCompleteSync` call, so it needs to be after the actual API call.  We follow this same pattern for all Signaling API calls.